File: Kpd37Gomel/Kpd37Gomel/ClientApp/src/app/apartment.ts
import {ITenant} from "./tenant";

export interface IApartment {
  id: string;
  apartmentNumber: string;
  floorNumber: number;
  totalAreaSnb: number;
  totalArea: number;
  livingArea: number;
  voteRate: number;
  owner: ITenant;
  tenants: Array<ITenant>;
}

export class Apartment implements IApartment {
  constructor(
    public id: string,
    public apartmentNumber: string,
    public floorNumber: number,
    public totalAreaSnb: number,
    public totalArea: number,
    public livingArea: number,
    public voteRate: number,
    public owner: ITenant,
    public tenants: Array<ITenant>) {}
}

File: Kpd37Gomel/Kpd37Gomel.DataAccess/Models/Apartment.cs
using System;
using System.Collections.Generic;

namespace Kpd37Gomel.DataAccess.Models
{
    public class Apartment
    {
        public Guid Id { get; set; }
        public int ApartmentNumber { get; set; }
        public int? FloorNumber { get; set; }
        public double? TotalAreaSnb { get; set; }
        public double TotalArea { get; set; }
        public double? LivingSpace { get; set; }
        public double VoteRate { get; set; }
        public int? RoomsCount { get; set; }
        public bool IsDeleted { get; set; }
        public DateTime? DeletionDateUtc { get; set; }

        public virtual ICollection<Tenant> Tenants { get; set; } = new List<Tenant>();
        public virtual ICollection<ApartmentVoteChoice> ApartmentVoteChoices { get; set; } = new List<ApartmentVoteChoice>();
    }
}
